cro-clib/src/transaction.rs

```diff
@@ -0,0 +1,318 @@
+    let from_address = from_ptr.as_mut().expect("get address");
+
+    match do_cro_unbond(network, nonce, from_address, &to_address, amount) {
+        Ok(encoded) => {
+            ptr::copy_nonoverlapping(encoded.as_ptr(), output, encoded.len());
+            (*output_length) = encoded.len() as u32;
+
+            CroResult::success()
+        }
+        Err(_) => CroResult::fail(),
+    }
+}
+
+fn do_cro_withdraw(
+    tendermint_url: &str,
+    network: u8,
+    from_address: &CroAddress,
+    to_address: &CroAddress,
+    viewkeys: Vec<String>, // viewkeys
+) -> Result<Vec<u8>> {
+    let from_private = &from_address.privatekey;
+    assert!(20 == from_address.raw.len());
+    let tendermint_client = WebsocketRpcClient::new(&tendermint_url)?;
+    let bytes = tendermint_client
+        .query("account", &from_address.raw)?
+        .bytes()?;
+    let staked_state = StakedState::decode(&mut bytes.as_slice()).chain(|| {
+        (
+            ErrorKind::DeserializationError,
+            format!(
+                "Cannot deserialize staked state for address: {}",
+                hex::encode(bytes)
+            ),
+        )
+    })?;
+    let mut access_policies = BTreeSet::new();
+    for a in viewkeys {
+        let view_key = a.trim();
+        let publickey = PublicKey::from_str(view_key)?;
+        access_policies.insert(TxAccessPolicy {
+            view_key: publickey.into(),
+            access: TxAccess::AllData,
+        });
+    }
+    let attributes = TxAttributes::new_with_access(network, access_policies.into_iter().collect());
+    let nonce = staked_state.nonce;
+    let amount = staked_state.unbonded;
+    let outputs = vec![TxOut::new_with_timelock(
+        ExtendedAddr::from_str(&to_address.address).chain(|| {
+            (
+                ErrorKind::DeserializationError,
+                format!("Unable to deserialize to_address ({})", to_address.address),
+            )
+        })?,
+        amount,
+        staked_state.unbonded_from,
+    )];
+    let transaction = WithdrawUnbondedTx::new(nonce, outputs, attributes);
+    let signature = from_private
+        .sign(transaction.id())
+        .map(StakedStateOpWitness::new)?;
+    let signed_transaction = SignedTransaction::WithdrawUnbondedStakeTransaction(
+        transaction,
+        Box::new(staked_state),
+        signature,
+    );
+    let encoded = signed_transaction.encode();
+    Ok(encoded)
+}
+
+/// staked -> utxo
+/// tendermint_url_string:  "ws://localhost:26657/websocket"
+/// network: network-id 0xab
+/// from_ptr: staking address
+/// to_ptr: transfer address
+/// viewkeys: viewkey list, this is string list
+/// output: minimum 1000 bytes, signed tx encoded
+#[no_mangle]
+/// # Safety
+pub unsafe extern "C" fn cro_withdraw(
+    tenermint_url_string: *const c_char,
+    network: u8,
+    from_ptr: CroAddressPtr,
+    to_ptr: CroAddressPtr,
+    viewkeys: *const *const c_char, // viewkeys
+    viewkey_count: i32,
```

Comment:
total size of the the array or the the number of items (as it can be calculated if each item is assumed to have the same size)? 
should be clarified

Reply:
i'll add more explanation, it's count of viewkeys.
viewkey is null-terminated strings.